bundles/org.openhab.binding.smaenergymeter/src/main/java/org/openhab/binding/smaenergymeter/internal/handler/SMAEnergyMeterHandler.java

```diff
@@ -61,68 +74,121 @@ public void initialize() {
 
         EnergyMeterConfig config = getConfigAs(EnergyMeterConfig.class);
 
-        int port = (config.getPort() == null) ? EnergyMeter.DEFAULT_MCAST_PORT : config.getPort();
-        energyMeter = new EnergyMeter(config.getMcastGroup(), port);
+        int port = (config.getPort() == null) ? PacketListener.DEFAULT_MCAST_PORT : config.getPort();
+
         try {
-            energyMeter.update();
+            PacketListener thingListener = listenerRegistry.getListener(config.getMcastGroup(), port);
+            serialNumber = config.getSerialNumber();
+            if (serialNumber == null) {
+                if (!thing.getProperties().containsKey(Thing.PROPERTY_SERIAL_NUMBER)
+                        || (serialNumber = retrieveSerialNumber(thing.getProperties())) == null) {
```

Comment:
No need to check in properties, serialNumber  will be set by the new configuration parameter you added to the thing.

Normally, the recent core framework should check that `config.getSerialNumber()` is not null for you as you defined the parameter as required. But keep you can keep the check.